### ISSUE TITLE
 org.apache.spark.sql.catalyst.expressions.BoundReference cannot be cast to class org.apache.spark.sql.ColumnarExpression

### ISSUE THREAD

**Issue by kazuyukitanimura** [apache/datafusion-comet#1197]
### Describe the bug

In SparkSessionExtensionSuite, there are 4 test failures with  `org.apache.spark.sql.catalyst.expressions.BoundReference cannot be cast to class org.apache.spark.sql.ColumnarExpression`
when `spark.comet.enabled=true` is set.

inject columnar AQE on
inject columnar AQE off
SPARK-39991: AQE should retain column statistics from completed query stages
reset column vectors

### Steps to reproduce

SparkSessionExtensionSuite is not using SharedSparkSessionBase, so add `spark.comet.enabled=true` by changing code

**Comment by andygrove** [comment]
@kazuyukitanimura Is this issue still valid?

**Comment by kazuyukitanimura** [comment]
@andygrove 
Yes, I think so https://github.com/apache/datafusion-comet/blob/e7a3214510091cf9d177fabdf9e3221317d1e785/dev/diffs/3.4.3.diff#L981
